### DISCUSSION TITLE
Powershell Deploy

### DISCUSSION THREAD

**Discussion by mycanaletto** [rustdesk/rustdesk#5828] [General]
### Bug Description

When you deploy the client with a script, you can adjust the permanent password and retrieve the ID

```
.\RustDesk.exe --get-id | more

.\RustDesk.exe --config $rustdesk_cfg

.\RustDesk.exe --password $rustdesk_pw
```

This is all very well, but for it to be fully deployable, you need to be able to configure everything:
* Be able to assign the customer to an account and/or a group
* Be able to modify all parameters, especially Enable RDP Sharing (very useful for monitoring servers)

### How to Reproduce

N/A

### Expected Behavior

N/A

### Operating system(s) on local side and remote side

Windows Server

### RustDesk Version(s) on local side and remote side

1.2.2

### Screenshots

N/A

**Comment by dinger1986** [comment]
Hello, the majority of this can be assigned from https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/console/ 

Tokens can be used to assign users to devices and included in the scripts

**Reply by mycanaletto** [reply]
I tested it with this but it doesn't work

`PS:C:\.\RustDesk.exe  –assign –token "rapi_xxxxxxxxxxxxxxxxxxx=" –user_name “admin”`

**Reply by dinger1986** [reply]
did you put into RustDesk? also was powershell ran with administrative rights?

**Comment by mycanaletto** [comment]
Yes, PS in admin mode

**Reply by dinger1986** [reply]
snip from mine I just did

![image](https://github.com/rustdesk/rustdesk/assets/7244447/de458f42-0022-4089-bbbe-e83d08bf74be)

Snip of my token permissions

![image](https://github.com/rustdesk/rustdesk/assets/7244447/5f3f1ff3-fc87-4a39-b481-7ac1e3acea70)

**Comment by mycanaletto** [comment]
![image](https://github.com/rustdesk/rustdesk/assets/45459269/a4ca5e16-41b8-4b1e-9b3e-5ebd677ba45a)

I dont have the : Done (permissions R/W

**Reply by dinger1986** [reply]
And still not assigned?

**Comment by mycanaletto** [comment]
no...
![image](https://github.com/rustdesk/rustdesk/assets/45459269/96b3b575-6c35-436e-ab80-728b4045c703)

**Comment by mycanaletto** [comment]
It is done in CMD (Admin) but not in PS (Admin) ....

```
PS C:\Program Files\RustDesk> .\RustDesk.exe  -–assign -–token "rapi_gKxxxxxxxxxxxxxxxxx==" -–user_name “admin” | more

```
```
C:\Program Files\RustDesk>RustDesk.exe  --assign --token "rapi_gKxxxxxxxxxxxxxxxxxx==" --user_name "admin" | more

Done!
```

**Reply by rustdesk** [reply]
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/out-host?view=powershell-7.3

**Reply by mycanaletto** [reply]
OK, thanks

`.\RustDesk.exe  --assign --token "rapi_xxxxxxxxxxxxxxxxx==" --user_name admin | out-host`
